warehouse/tuf/services.py

```diff
@@ -0,0 +1,408 @@
+            ),
+            store=True,
+        )
+
+    def bump_bins_ns(self):
+        """
+        Bump all BIN-S delegate roles Metadata
+        """
+        # Bumping all of the delegated bin roles in the TUF repository involves
+        # the following steps:
+        # 1. Metadata Repository.
+        # 2. Load Snapshot role.
+        # 3. Load BIN-S role.
+        # 4. For each delegated hashed bin targets role in the BIN-S, fetch the
+        #    role, bump, write back to the repo and update Snapshot role MetaFile.
+        # 5. Bump BIN-S and write back to repository.
+        # 6. Bump Snapshot and write back to repository.
+        # 7. Bump Timestamp role (using updated Snapshot) and write back to
+        #    repository.
+
+        # 1. Metadata Repository
+        metadata_repository = MetadataRepository(
+            self._storage_backend, self._key_storage_backend
+        )
+
+        # 2. Load Snapshot role.
+        snapshot_metadata = metadata_repository.load_role(Role.SNAPSHOT.value)
+
+        # 3 Load BIN-S role.
+        bins_metadata = metadata_repository.load_role(Role.BINS.value)
+
+        # 4. Fore each delegated hashed bin target role, bump and update Snapshot
+        for role in bins_metadata.signed.delegations.roles.keys():
+            role_metadata = metadata_repository.load_role(role)
+            metadata_repository.bump_role_version(
+                rolename=role,
+                role_metadata=role_metadata,
+                role_expires=self._set_expiration_for_role(Role.BINS.value),
+                snapshot_metadata=snapshot_metadata,
+                key_rolename=Role.BIN_N.value,
+                store=True,
+            )
+            snapshot_metadata = metadata_repository.snapshot_update_meta(
+                role, role_metadata.signed.version, snapshot_metadata
+            )
+
+        # 5. Bump BIN-S with updated metadata
+        snapshot_metadata = metadata_repository.bump_role_version(
+            rolename=Role.BINS.value,
+            role_metadata=bins_metadata,
+            role_expires=self._set_expiration_for_role(Role.BINS.value),
+            snapshot_metadata=snapshot_metadata,
+            key_rolename=None,
+            store=True,
+        )
+
+        # 6. Bump Snapshot with updated metadata
+        snapshot_metadata = metadata_repository.snapshot_bump_version(
+            snapshot_expires=self._set_expiration_for_role(
+                self._request, Role.SNAPSHOT.value
+            ),
+            snapshot_metadata=snapshot_metadata,
+            store=True,
+        )
+
+        # Bump Timestamp with updated Snapshot metadata
+        metadata_repository.timestamp_bump_version(
+            snapshot_version=snapshot_metadata.signed.version,
+            timestamp_expires=self._set_expiration_for_role(
+                self._request, Role.TIMESTAMP.value
+            ),
+            store=True,
+        )
+
+    def delegate_targets_bin_bins(self):
```

Comment:
This function bootstraps the targets delegation tree from the "top-level targets" role to the  "bins" role to all "bin-n" roles, creating initial metadata for bins and bin-n roles. I think the name does not capture that well. 

I prefer the more general name used in the cli `delegate_targets_roles`.

Also see comment about bin/bins/bin-n/bin-ns/... name disambiguation in cli/tuf.py (line 89).

Reply:
Given that this is a bootstrapping command, would it make sense to express that in the name? E.g.
```suggestion
    def init_targets_delegation(self):
```

Reply:
And is there a reason to not just combine this with `init_repository` above?